src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs

```diff
@@ -252,6 +252,19 @@ public void Checkout(string repoDir, string commit, bool force = false)
         }
     }
 
+    public async Task CheckoutNativeAsync(string repoDir, string refToCheckout, bool createBranch = false, bool overwriteExistingBranch = false)
```

Comment:
IMO this should be two methods. There is only a single use as the branch method. Maybe CheckoutRefNativeAsync and CreateBranchAsync?

Reply:
Oh yeah, good point